samcli/local/docker/container.py

```diff
@@ -314,6 +334,32 @@ def wait_for_logs(self, stdout=None, stderr=None):
 
         self._write_container_output(logs_itr, stdout=stdout, stderr=stderr)
 
+    def wait_for_socket_connection(self):
+        """
+        Waits for a successful connection to the socket used to communicate with Docker.
+        """
+        start_time = time.time()
+        sleep = 0.1
+        while True:
```

Comment:
Should we wait for `is_port_open` instead?

Reply:
we have a `break` inside this loop which should accomplish the same thing

Reply:
My general practice will be waited on a variable instead of `True` as waiting on `True` have a higher chance to lead to infinite loop execution than a variable. This is not critical but I'd generally advocate not to use a constant in a loop if possible.

Reply:
done 👍